src/App.tsx

```diff
@@ -26,6 +28,21 @@ export default function App() {
     />
   );
 
+  const filter = useCallback(
```

Comment:
The only time you really want to use `useCallback` is if it is being passed to a `PureComponent` (or component wrapped in `React.memo`). This really isn't needed.

Reply:
I was under the impression that you generally wanted stable props to prevent unnecessary re-renders. Though, in this case, it's a fairly simple component, so you probably lose the benefit with the overhead of `useCallback`. Will remove.